spec/integration/bisect_spec.rb

```diff
@@ -45,7 +45,10 @@ def bisect(cli_args, expected_status=nil)
       it 'does not leave zombie processes', :unless => RSpec::Support::OS.windows? do
         original_pids = pids()
         bisect(%W[spec/rspec/core/resources/blocking_pipe_bisect_spec.rb_], 1)
+        cursor = 0
         while ((extra_pids = pids() - original_pids).join =~ /[RE]/i)
+          raise "Extra process detected" if cursor > 10
```

Comment:
What could be cleaner than raising here?

Reply:
Also 10 means nothing. If we wait more than 1s then we raise an error. Is it clear enough?

Otherwise I think the PR can be merged @JonRowe

Reply:
We really just need a wait of finding the bisect pid, can we grep for the rspec command, does that mean we'll have less processes to check?

Reply:
It requires a little bit more work to find child processes. I have submitted a patch that does that. https://github.com/rspec/rspec-core/pull/2739/commits/0774f41186ae7cb6009212c927443a9446b6efa7

Without the patch in `fork_runner`. The actual code returns:

```
Bisect
  when the bisect command saturates the pipe
    does not hit pipe size limit and does not get stuck
    does not leave zombie processes (FAILED - 1)
  when the spec ordering is inconsistent
    stops bisecting and surfaces the problem to the user
  when a load-time problem occurs while running the suite
    surfaces the stdout and stderr output to the user

Failures:

  1) Bisect when the bisect command saturates the pipe does not leave zombie processes
     Failure/Error:
               expect(zombie_process).to eq([]), <<-MSG
                 Expected no zombie processes got #{zombie_process.count}:
                   #{zombie_process}
               MSG

                 Expected no zombie processes got 2:
                   [#<struct RSpec::Core::RSpecChildProcess::Ps pid="31512", ppid="31503", state="Z+", command="(ruby)">, #<struct RSpec::Core::RSpecChildProcess::Ps pid="31513", ppid="31503", state="Z+", command="(ruby)">]
````

Reply:
I had no random failures. So I removed the part where we had to wait for the process execution.